Fix provide at least one entry - Phabricator or Gerrit Username

rammanoj opened the pull request:
Fix provide at-least one entry for phabricator and gerrit username and Full Name field left blank on the main page results in server loading forever

Fix #50, #49

rammanoj commented:
@srish It would be great if you can check it once locally :)

srish commented on frontend/WikiContrib-Frontend/src/query.js:
```diff
@@ -281,7 +289,33 @@ export class Query extends Component {
+      this.setState({
+        message: {
+          message:
+            'Fill either gerrit or phabricator username at row ' +
```
Wording: "Both Gerrit and Phabricator fields cannot be left blank. Provide username for one of these accounts in row +  .... !"

rammanoj commented on backend/Contraband/result/views.py:
```diff
@@ -248,35 +256,39 @@ def get(self, request, *args, **kwargs):
+                        user = file.iloc[ind, :]
+
+                    if not temp:
+                        return Response({'message': 'Fill file with proper entities', 'error': 1}, status=status.HTTP_404_NOT_FOUND)
```
This error is raised when it is missing for all the users (I even assumed the worst case, that may never occur). So renaming the sentence to "Full name and both Gerrit and Phabricator username cannot be left blank. It is missing for **all** user(s) in the CSV file!"

rammanoj commented on backend/Contraband/result/views.py:
```diff
@@ -248,35 +256,39 @@ def get(self, request, *args, **kwargs):
-                    return Response({'message': 'CSV file is not in specified format!!', 'error': 1},
-                                    status=status.HTTP_400_BAD_REQUEST)
             except FileNotFoundError:
                 return Response({'message': 'Not Found', 'error': 1}, status=status.HTTP_404_NOT_FOUND)
```
Nope. This is my mistake. I removed try catch block for debugging and forgot to add it back. Thanks for pointing it out :)

rammanoj commented on backend/Contraband/result/views.py:
```diff
@@ -248,35 +256,39 @@ def get(self, request, *args, **kwargs):
+                if 'user' in request.GET:
+                    user = file[file['fullname'] == request.GET['user']]
+                    if user.empty:
+                        return Response({'message': 'User not found', 'error': 1}, status=status.HTTP_404_NOT_FOUND)
```
This does not show any affect over the app because the frontend of the app is programmed in such a manner so that the it never raises the error. You might have observed the pagination in the app. The error is raised when any developer tries to access the API of the tool and provides with the username that is not in the CSV file.

rammanoj commented on frontend/WikiContrib-Frontend/src/api.js:
```diff
@@ -2,8 +2,8 @@
-const BASE_API_URI = 'https://tools.wmflabs.org/contraband/';
-// const BASE_API_URI = "http://127.0.0.1:8000/";
+// const BASE_API_URI = 'https://tools.wmflabs.org/contraband/';
+const BASE_API_URI = 'http://127.0.0.1:8000/';
```
Yeah!! Thanks for pointing out :+1:

rammanoj commented:
@srish I've updated with all the changes you specified. Thanks a lot for the review. I also added info icon using &#9432; (In HTML it generates an info icon). Refer https://stackoverflow.com/questions/33878539/is-there-an-html-entity-for-an-info-icon

rammanoj commented:
The server will be updated with all the changes by tomorrow :)
